components/gitpod-cli/pkg/utils/trackEvent.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package utils
+
+import (
+	"context"
+	"time"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-cli/pkg/gitpod"
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor"
+	serverapi "github.com/gitpod-io/gitpod/gitpod-protocol"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// Rebuild
+	RebuildErrorCode_MalformedGitpodYaml = "rebuild_malformed_gitpod_yaml"
+	RebuildErrorCode_MissingGitpodYaml   = "rebuild_missing_gitpod_yaml"
+	RebuildErrorCode_DockerfileNotFound  = "rebuild_dockerfile_not_found"
+	RebuildErrorCode_DockerfileEmpty     = "rebuild_dockerfile_empty"
+	RebuildErrorCode_DockerNotFound      = "rebuild_docker_not_found"
+	RebuildErrorCode_DockerBuildFailed   = "rebuild_docker_build_failed"
+	RebuildErrorCode_DockerRunFailed     = "rebuild_docker_run_failed"
+	RebuildErrorCode_DockerErr           = "rebuild_docker_err"
+)
+
+type TrackCommandUsageParams struct {
+	Command             string `json:"command,omitempty"`
+	DurationMs          int64  `json:"durationMs,omitempty"`
+	Success             bool   `json:"success,omitempty"`
+	ErrorCode           string `json:"errorCode,omitempty"`
+	WorkspaceId         string `json:"workspaceId,omitempty"`
+	WorkspaceInstanceId string `json:"workspaceInstanceId,omitempty"`
+	Timestamp           int64  `json:"timestamp,omitempty"`
+}
+
+type EventTracker struct {
+	data             *TrackCommandUsageParams
+	startTime        time.Time
+	serverClient     *serverapi.APIoverJSONRPC
+	supervisorClient *supervisor.SupervisorClient
+}
+
+func TrackEvent(ctx context.Context, supervisorClient *supervisor.SupervisorClient, cmdParams *TrackCommandUsageParams) *EventTracker {
+	tracker := &EventTracker{
+		startTime:        time.Now(),
+		supervisorClient: supervisorClient,
+	}
+
+	wsInfo, err := supervisorClient.Info.WorkspaceInfo(ctx, &api.WorkspaceInfoRequest{})
+	if err != nil {
+		LogError(ctx, err, "Could not fetch the workspace info", supervisorClient)
+		return nil
+	}
+
+	serverClient, err := gitpod.ConnectToServer(ctx, wsInfo, []string{"function:trackEvent"})
```

Comment:
Can’t we report directly to segment untrusted.

Mae are trying to get rid to of connection to server.

Reply:
so, I hit a road block here, couldn't figure out why Segment won't show the data, so I've reverted to using server. We can refactor this and the other other occurrences in gitpod-cli with this issue: https://github.com/gitpod-io/gitpod/issues/10349

Reply:
We decided to keep this change to get this over the line. It's not ideal as we do want to get rid of the connection to server, but we can open an issue so we have it scheduled. For now let's get the data coming in with this skateboard and continue iterating on it  💪🏼